Sources/PackageGraph/PackageGraph+Loading.swift

```diff
@@ -514,9 +557,21 @@ private func createResolvedPackages(
             }
         }
     }
+
+    if moduleAliasingUsed {
+        // Since duplicate product names are allowed for automatic or static lib products,
+        // they need to be renamed as IDs here.
+        productList.filter{ $0.isStaticLibrary }.forEach { $0.useIDAsName() }
+    }
     return try packageBuilders.map{ try $0.construct() }
 }
 
+fileprivate extension Product {
+    var isStaticLibrary: Bool {
+        return type == .library(.automatic) || type == .library(.static)
```

Comment:
@neonichu does this match your expectation?

Reply:
modified to auto only